platform/commonUI/notification/src/NotificationService.js

```diff
@@ -0,0 +1,284 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT Web includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+/*global define*/
+
+/**
+ * This bundle implements the notification service, which can be used to
+ * show banner notifications to the user. Banner notifications
+ * are used to inform users of events in a non-intrusive way. As
+ * much as possible, notifications share a model with blocking
+ * dialogs so that the same information can be provided in a dialog
+ * and then minimized to a banner notification if needed.
+ *
+ * @namespace platform/commonUI/dialog
+ */
+define(
+    ["./MessageSeverity"],
+    function (MessageSeverity) {
+        "use strict";
+
+        /**
+         * A representation of a user action. Actions are provided to
+         * dialogs and notifications and are shown as buttons.
+         *
+         * @typedef {object} NotificationAction
+         * @property {string} label the label to appear on the button for
+         * this action
+         * @property {function} action a callback function to be invoked
+         * when the button is clicked
+        */
+
+        /**
+         * A representation of a banner notification. Banner notifications
+         * are used to inform users of events in a non-intrusive way. As
+         * much as possible, notifications share a model with blocking
+         * dialogs so that the same information can be provided in a dialog
+         * and then minimized to a banner notification if needed.
+         *
+         * @typedef {object} Notification
+         * @property {string} title The title of the message
+         * @property {MessageSeverity} severity The importance of the
+         * message (eg. error, info)
+         * @property {number} progress The completion status of a task
+         * represented numerically
+         * @property {boolean} unknownProgress a boolean indicating that the
+         * progress of the underlying task is unknown. This will result in a
+         * visually distinct progress bar.
+         * @property {boolean | number} autoDismiss If truthy, dialog will
+         * be automatically minimized or dismissed (depending on severity).
+         * Additionally, if the provided value is a number, it will be used
+         * as the delay period before being dismissed.
+         * @property {NotificationAction} primaryAction the default user
```

Comment:
Similar to above, would prefer to call this a `NotificationOption` to avoid confusion with `Action` types.